runtime/cudaq/algorithms/gradients/parameter_shift.h

```diff
@@ -16,6 +16,12 @@ class parameter_shift : public gradient {
   using gradient::gradient;
   double shiftScalar = 0.5;
 
+  virtual std::unique_ptr<cudaq::gradient> clone() override {
+    auto newGrad = std::make_unique<parameter_shift>(*this);
+    newGrad->shiftScalar = this->shiftScalar;
+    return newGrad;
+  }
+
```

Comment:
No, I don't think so. The `gradient` class is already an abstract base class, but it doesn't know about the derived class's member variables, so we cannot put the implementation in there. The implementations are different for each of the subclassed clone methods so that they can handle their own class variables.